app/controllers/new-signup.js

```diff
@@ -91,10 +122,14 @@ export default class NewSignUpController extends Controller {
   }
 
   @action async signup() {
+    const user = await this.generateUsername(
+      this.signupDetails.firstName,
+      this.signupDetails.lastName
+    );
     const signupDetails = {
       first_name: this.signupDetails.firstName,
       last_name: this.signupDetails.lastName,
-      username: this.signupDetails.username,
+      username: user?.username,
```

Comment:
User cannot be optional

Reply:
how and when will the users be optional

Reply:
In case of the API error, it is possible that user is undefined.

Reply:
If the users is undefined should we come till this flow?

Reply:
> If the users is undefined should we come till this flow?

for me no @lakshayman

Reply:
Updated the code, according to it. Please check @iamitprakash @prakashchoudhary07